StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1117ParametersMustBeOnSameLineOrSeparateLines.cs

```diff
@@ -290,81 +218,44 @@ private static void HandleParameterListSyntax(SyntaxNodeAnalysisContext context,
-            if (parameters.Count > 2)
-            {
-                Analyze(context, parameters);
-            }
+            Analyze(context, parameters);
         }
 
         private static void HandleBracketedArgumentListSyntax(SyntaxNodeAnalysisContext context, BracketedArgumentListSyntax bracketedArgumentList)
         {
             SeparatedSyntaxList<ArgumentSyntax> arguments = bracketedArgumentList.Arguments;
-            if (arguments.Count > 2)
-            {
-                Analyze(context, arguments);
-            }
+            Analyze(context, arguments);
         }
 
-        private static void Analyze(SyntaxNodeAnalysisContext context, SeparatedSyntaxList<ParameterSyntax> parameters)
+        private static void Analyze(SyntaxNodeAnalysisContext context, IReadOnlyList<SyntaxNode> arguments)
         {
-            ParameterSyntax previousParameter = parameters[1];
-            int firstParameterLine = parameters[0].GetLine();
-            int previousLine = previousParameter.GetLine();
-            Func<int, int, bool> lineCondition;
-
-            if (firstParameterLine == previousLine)
+            if (arguments.Count < 2)
             {
-                // parameters should be on same line
-                lineCondition = (param1Line, param2Line) => param1Line == param2Line;
-            }
-            else
-            {
-                // each parameter should be on its own line
-                lineCondition = (param1Line, param2Line) => param1Line != param2Line;
-            }
-
-            for (int i = 2; i < parameters.Count; ++i)
-            {
-                ParameterSyntax currentParameter = parameters[i];
-                int currentLine = currentParameter.GetLine();
-
-                if (lineCondition(previousLine, currentLine))
-                {
-                    previousLine = currentLine;
-                    continue;
-                }
-
-                context.ReportDiagnostic(Diagnostic.Create(Descriptor, currentParameter.GetLocation()));
                 return;
             }
-        }
 
-        private static void Analyze(SyntaxNodeAnalysisContext context, SeparatedSyntaxList<ArgumentSyntax> arguments)
-        {
-            ArgumentSyntax previousParameter = arguments[1];
-            int firstParameterLine = arguments[0].GetLine();
-            int previousLine = previousParameter.GetLine();
-            Func<int, int, bool> lineCondition;
+            SyntaxNode firstParameter = arguments[0];
+            SyntaxNode secondParameter = arguments[1];
+            Func<SyntaxNode, SyntaxNode, bool> lineCondition;
 
-            if (firstParameterLine == previousLine)
+            if (firstParameter.GetLine() == secondParameter.GetLine())
             {
                 // arguments should be on same line
-                lineCondition = (param1Line, param2Line) => param1Line == param2Line;
+                lineCondition = (param1, param2) => param1.GetLine() == param2.GetEndLine();
```

Comment:
📝 This one I'm not so sure about. For now I'm going to relax the condition:

```suggestion
                lineCondition = (param1, param2) => param1.GetLine() == param2.GetLine();
```

This change means we still fix #3106, but it doesn't address #3183 which is still in discussion.

Reply:
➡️ This one is now changed back to only compare starting lines.